src/v/cluster/data_migration_types.h

```diff
@@ -384,4 +430,38 @@ struct remove_migration_reply
     operator<<(std::ostream&, const remove_migration_reply&);
 };
 
+struct check_ntp_states_request
+  : serde::envelope<
+      check_ntp_states_request,
+      serde::version<0>,
+      serde::compat_version<0>> {
+    using rpc_adl_exempt = std::true_type;
+    using self = check_ntp_states_request;
+
+    fragmented_vector<data_migration_ntp_state> sought_states;
```

Comment:
please use a `chunked_vector` type to benefit from its adaptive chunk size

Reply:
do we have any kind of rule of thumb to decide between the two (and other containers)?

Reply:
in general we should use `chunked_vector`.

Reply:
or an `std::vector if collection is really small, say up to a dozen elements